File: RecyclerViewJava/app/src/main/java/com/devwilly/recyclerview_java/viewholders/FoodViewHolder.java
package com.devwilly.recyclerview_java.viewholders;

import com.devwilly.recyclerview_java.R;
import com.devwilly.recyclerview_java.RecyclerViewAdapter;
import com.devwilly.recyclerview_java.wrapper.FoodWrapper;

import android.view.View;
import android.widget.TextView;


/**
 * Created by Willy on 15/01/2018.
 */

public class FoodViewHolder extends BaseViewHolder<FoodWrapper> {

    private TextView mFoodName;
    private TextView mFoodDescription;
    private TextView mFoodPrice;
    private RecyclerViewAdapter.OnItemClickListener mListener;

    public FoodViewHolder(View itemView, RecyclerViewAdapter.OnItemClickListener listener) {
        super(itemView);
        this.mListener = listener;

        mFoodName = itemView.findViewById(R.id.food_name);
        mFoodDescription = itemView.findViewById(R.id.food_description);
        mFoodPrice = itemView.findViewById(R.id.food_price);
    }

    @Override
    public void bindView(final FoodWrapper wrapper) {
        mFoodName.setText(wrapper.getName());
        mFoodDescription.setText(wrapper.getDescription());
        mFoodPrice.setText(wrapper.getPrice());

        itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                mListener.onFoodItemClick(wrapper.getName());
            }
        });
    }
}

File: RecyclerView-Kotlin/app/src/main/java/com/devwilly/recyclerviewkotlin/viewholders/FoodViewHolder.kt
package com.devwilly.recyclerviewkotlin.viewholders

import android.view.View
import android.widget.TextView
import com.devwilly.recyclerviewkotlin.R
import com.devwilly.recyclerviewkotlin.RecyclerViewAdapter
import com.devwilly.recyclerviewkotlin.wrapper.IEatingWrapper

/**
 * Created by Willy on 15/01/2018.
 */
class FoodViewHolder(itemView: View, listener: RecyclerViewAdapter.OnItemClickListener?) : BaseViewHolder<IEatingWrapper>(itemView) {

    private val mListener = listener
    private val mFoodName = itemView.findViewById<TextView>(R.id.food_name)
    private val mFoodDescription = itemView.findViewById<TextView>(R.id.food_description)
    private val mFoodPrice = itemView.findViewById<TextView>(R.id.food_price)

    override fun bindView(t: IEatingWrapper) {
        mFoodName.text = t.getName()
        mFoodDescription.text = t.getDescription()
        mFoodPrice.text = t.getPrice()

        itemView.setOnClickListener { mListener?.onFoodItemClick(t.getName()) }
    }
}